datalad/distribution/dataset.py

```diff
@@ -567,10 +567,14 @@ def require_dataset(dataset, check_installed=True, purpose=None):
         dspath = get_dataset_root(getpwd())
         if not dspath:
             raise NoDatasetFound(
-                "No dataset found at '{}'.  Specify a dataset to work with "
+                "No dataset found at '{}'{}.  Specify a dataset to work with "
                 "by providing its path via the `dataset` option, "
                 "or change the current working directory to be in a "
-                "dataset.".format(getpwd()))
+                "dataset.".format(
+                    getpwd(),
+                    " for the purpose {!r}".format(purpose) if purpose else ''
```

Comment:
With your harmonization of purposes, this wording feels wrong.
"for the purpose report difference"? May be just "to {!r}" ?

Reply:
Alternatively the "ing" form may be easier to handle here. But feeling bad proposing this after you've done it the other way around throughout the code base ;-)

Reply:
I agree that "to {!r}" would be much easier to comprehend in this example

Reply:
yeap -- if #5733 is accepted (before), I would need to tune it here as well to the shorter
```suggestion
                    " to {!r}".format(purpose) if purpose else ''
```
you suggest ;-)